lib/http_server.ex

```diff
@@ -51,57 +53,12 @@ defmodule HTTPServer do
     socket
   end
 
-  def parse_request(message) do
-    [request_data | body] = message |> String.split("\r\n\r\n")
-    [first | headers] = request_data |> String.split("\r\n")
-    [method, path, resource] = first |> String.split(" ")
+  def determine_routes do
+    env = Mix.env()
 
-    req = %Request{
-      method: method,
-      path: path,
-      resource: resource,
-      headers: parse_headers(headers, %{}),
-      body: hd(body)
-    }
-
-    case method do
-      "POST" ->
-        %{req | method: :post}
-
-      _ ->
-        {:error, :unknown_command}
+    case env do
+      :test -> &HTTPServerFixture.Routes.route/1
+      _ -> &Routes.route/1
```

Comment:
I like the creativity of this solution.  though I would point out that there should be an explicit difference between something like `:test` (which connotes unit tests) and `:integration_test` (which would be the http_server_specs).
```suggestion
    case env do
      :test -> &HTTPServer.TestRoutes.route/1
      :integration_test -> &HTTPServerFixture.Routes.route/1
      _ -> &Routes.route/1
```

Reply:
also, there appears to be configuration code here, I wonder if this is the best place for this logic?  I am not really sure, so that's food for thought for a future PR.